bletl/growth.py

```diff
@@ -170,29 +181,51 @@ def _make_random_walk(name:str, *, mu: float=0, sigma:float, nu: float=1, length
     length : int
         Number of steps in the random walk.
     student_t : bool
-        If `True` a `pymc3.Deterministic` of a StudentT-random walk is created.
+        If `True` a `pymc.Deterministic` of a StudentT-random walk is created.
         Otherwise a `GaussianRandomWalk` is created.
     initval : numpy.ndarray
         Initial values for the RandomWalk variable.
-        If set, PyMC3 uses these values as start points for MAP optimization and MCMC sampling.
+        If set, PyMC uses these values as start points for MAP optimization and MCMC sampling.
 
     Returns
     -------
     random_walk : TensorVariable
         The tensor variable of the random walk.
     """
-    if student_t:
-        # a random walk of length N is just the cumulative sum over a N-dimensional random variable:
-        return pymc3.Deterministic(name, tt.cumsum(
-            pymc3.StudentT(
-                f'{name}__diff_',
-                mu=mu, sd=sigma, nu=nu,
-                shape=(length,),
-                testval=numpy.diff(initval, prepend=0) if initval is not None else initval
-            )
-        ))
+    pmversion = version.parse(pm.__version__)
+
+    # Adapt to rename of the testval→initval kwarg
+    if pmversion <= version.parse("3.11.4"):
+        initval_kwarg = "testval"
     else:
-        return pymc3.GaussianRandomWalk(name, mu=mu, sigma=sigma, shape=(length,), testval=initval)
+        initval_kwarg = "initval"
+
+    if pmversion < version.parse("4.0.0b1") and not student_t:
+        # Use the gaussian random walk distribution directly.
+        return pm.GaussianRandomWalk(**{
+            "name": name,
+            "mu": mu,
+            "sigma": sigma,
+            "shape": (length,),
+            initval_kwarg: initval,
+        })
+    else:
+        # Create the random walk manually.
+        rv_kwargs = {
+            "name": f"{name}__diff_",
+            "mu": mu,
+            "sigma": sigma,
+            "shape": (length,),
+            initval_kwarg: numpy.diff(initval, prepend=0) if initval is not None else None,
```

Comment:
Why the` numpy.diff`?

Reply:
The `initval` passed from the outside refers to the final random walk, but the actual RV created here is just a Normal/StudentT which is `cumsum`ed.
So the initial value for the actual RV is the inverse of the cumsum operation: `numpy.diff`.

I'll add an explanatory comment.